### PR TITLE
Fetch Wiring Updates

### PR DESCRIPTION
**Pull request by cfazzini**
Some updates to pull in all pages of the wiring diagrams and also to get all the Connector view/pinouts.

### CODE REVIEW

File: src/wiring/fetchConnectPage.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+export default async function fetchConnectorPage(
+  params: FetchConnectorParams,
+  browserPage: Page,
+  saveImagePath: string
+): Promise<void> {
+  const url = `https://www.fordtechservice.dealerconnection.com/Wiring/face/?${stringify(
+    params
+  )}`;
+  console.log(url);
```
Comment (iamtheyammer) [inline src/wiring/fetchConnectPage.ts:23]: No worries in leaving a log throughout the code-- I have a couple-- but would you mind formatting the URL, something like 
```suggestion
  console.log(`Downloading wiring connector page ${url}`);
```

File: src/wiring/fetchConnectorList.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  params: FetchConnectorList,
+  cookieString: string,
+): Promise<ConnectorTableOfContents[]> {
+  const req = await client({
+    method: "GET",
+    url: "https://www.fordservicecontent.com/Ford_Content/PublicationRuntimeRefreshPTS//wiring/GetConnectorList",
+    params: {
+      ...params,
+      fromPageBase: "https://www.fordtechservice.dealerconnection.com",
+      page: 1
```
Comment (iamtheyammer) [inline src/wiring/fetchConnectorList.ts:29]: Do the pages change? Would we ever need page 2?

File: src/wiring/fetchPageList.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import client from "../client";
+
+export interface FetchWiringPageList {
+  book: string;
+  cell: string;
+  title: string;
+  bookType: string;
+  contentmarket: string;
+  contentlanguage: string;
+
```
Comment (iamtheyammer) [inline src/wiring/fetchPageList.ts:10]:
```suggestion
```

File: src/wiring/saveEntireWiring.ts [hunk 4]
```diff
@@ -26,34 +29,92 @@ export default async function saveEntireWiring(
       throw e;
     }
   }
+  const connecterPath = join(path, "Connectors");
```
Comment (iamtheyammer) [inline src/wiring/saveEntireWiring.ts:32]:
```suggestion
  const connectorPath = join(path, "Connectors");
```

File: src/wiring/saveEntireWiring.ts [hunk 5]
```diff
@@ -26,34 +29,92 @@ export default async function saveEntireWiring(
...
+  try {
+    await mkdir(connecterPath);
```
Comment (iamtheyammer) [inline src/wiring/saveEntireWiring.ts:34]:
```suggestion
    await mkdir(connectorPath);
```

File: src/wiring/saveEntireWiring.ts [hunk 6]
```diff
@@ -26,34 +29,92 @@ export default async function saveEntireWiring(
...
-
-    if (doc.Type === "Page") {
-      console.log(
-        `Downloading wiring manual page ${doc.Title} as image (#${doc.Number})`
-      );
-
-      await fetchPage(
+    // 150 = Connector views, different format handled later
+    // Always 150?
+    if (doc.Type === "Page" && doc.Number !== "150") {
```
Comment (iamtheyammer) [inline src/wiring/saveEntireWiring.ts:47]: Is it always 150, or is it 150 only for your car-- could you see if another car has its connectors as 150? My guess, although I very well could be wrong, is that the page numbers change per car.

**Review by iamtheyammer — CHANGES_REQUESTED** [review]
Looks pretty good-- I left a couple of questions and suggestions.

Also, could you run `yarn format`? That'll auto-format the code. Thanks so much for contributing code!

File: src/wiring/saveEntireWiring.ts (hunk 6 above)
Comment (cfazzini) [reply src/wiring/saveEntireWiring.ts:47]: It seems they are the same over a few vehicles.. switchin to use Title, should be better.

**Comment by cfazzini** [comment]
I think resolved all issues... my access just ran out so I'm not able to do a final retest. Though the logic/functions worked for a few different vehicles.

**Comment by iamtheyammer** [comment]
Sounds good. Chances are you can still access PTS if you go through your browser history and pick a service manual page, at least for a 24hr period.

**Comment by iamtheyammer** [comment]
Wiring fixes merged into main

**Pull request closed without merging**